Code/Source/Spawner/ROS2SpawnerComponent.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2SpawnerComponent.h"
+#include <AzCore/Math/Quaternion.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+
+namespace ROS2
+{
+    void ROS2SpawnerComponent::Activate()
+    {
+        auto ros2Node = ROS2Interface::Get()->GetNode();
+
+        m_get_names_service = ros2Node->create_service<gazebo_msgs::srv::GetWorldProperties>(
+            "get_available_spawnable_names",
```

Comment:
It is a good idea to include a configurable namespace here - does not cost much and adds support for multiple simulations (on network), or multiple service providers.

Reply:
Should it be configurable in the editor as a component field or in some other way?

Reply:
For now, just add an issue for this, I guess we can start with no namespace

Reply:
https://github.com/RobotecAI/o3de-ros2-gem/issues/204